cmd/otel-agent/config/agent_config.go

```diff
@@ -75,6 +105,22 @@ func NewConfigComponent(ctx context.Context, ddCfg string, uris []string) (confi
 		}
 	}
 
+	// Set the right log level. The most verbose setting takes precedence.
+	var logLevel string
+	agentLogLevel := pkgconfig.GetString("log_level")
+	telemetryLogLevel := sc.Telemetry.Logs.Level
+	agentLogMapping, ok := logLevelMap[agentLogLevel]
+	if ok {
+		logLevel = agentLogLevel
+	}
+	telemetryLogMapping, ok := logLevelMap[telemetryLogLevel.String()]
+	if ok {
+		if telemetryLogMapping.value > agentLogMapping.value {
+			logLevel = telemetryLogLevel.String()
+		}
+	}
+	pkgconfig.Set("log_level", logLevel, pkgconfigmodel.SourceLocalConfigProcess)
```

Comment:
is there ever a situation where logLevelMap wouldn't have a log_level value AND logLevelMap wouldn't also have a valid telemetryLogLevel value? is default loglevel setting handled elsewhere?